engine/src/main/java/org/hibernate/search/engine/backend/types/dsl/ScaledNumberIndexFieldTypeContext.java

```diff
@@ -16,4 +16,6 @@
 
 	S decimalScale(int decimalScale);
 
+	S defaultDecimalScale(int decimalScale);
```

Comment:
I think I mentioned in one of my previous commits that this default should be contributed by passing an object to `org.hibernate.search.engine.mapper.mapping.building.spi.IndexBindingContext#getTypeFactory`. There was a reason, actually: the type you're modifying here is API. If you add a `defaultDecimalScale` method here, users will be able to call it. I don't know about you, but I don't want to have to care about what happens if they call it twice, or if they call it and the ORM mapper also calls it, etc. So I'd rather not expose this feature to users.

If, instead, you change `org.hibernate.search.engine.mapper.mapping.building.spi.IndexBindingContext#getTypeFactory` to pass a `IndexFieldTypeDefaultsProvider` object, I think the result will be easier to test thoroughly.

Reply:
it does make sense!